src/WebJobs.Script/OutOfProc/Rpc/RpcWorkerDescription.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using Microsoft.Azure.WebJobs.Script.OutOfProc;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public class RpcWorkerDescription : WorkerDescription
+    {
+        private List<string> _extensions = new List<string>();
+
+        /// <summary>
+        /// Gets or sets the name of the supported language. This is the same name as the IConfiguration section for the worker.
+        /// </summary>
+        [JsonProperty(PropertyName = "language")]
+        public string Language { get; set; }
+
+        /// <summary>
+        /// Gets or sets the supported file extension type. Functions are registered with workers based on extension.
+        /// </summary>
+        [JsonProperty(PropertyName = "extensions")]
+        public List<string> Extensions
+        {
+            get
+            {
+                return _extensions;
+            }
+
+            set
+            {
+                if (value != null)
+                {
+                    _extensions = value;
+                }
+            }
+        }
+
+        public override void ApplyDefaultsAndValidate()
```

Comment:
Why not split `ApplyDefaults` and `Validate()` in 2 methods?

Reply:
before validating, defaults need to be applied. Splitting would be error prone if validate is called before applying defaults